### ISSUE TITLE
raft topology: enable writes to previous CDC generations

### ISSUE THREAD

**Issue by patjed41** [scylladb/scylladb#17184]
A follow-up issue for #17134. That PR enables writes to the previous CDC generations by changing `cdc::metadata`. However, these changes are sufficient only if the cluster uses the gossiper-based topology. For the Raft-based topology, there are two problems that need to be fixed.

The first problem is that we load only the current CDC generation from `cdc_generations_v3` in `storage_service::topology_state_load`. For example, a new node may not load a generation preceding the current generation, but it could receive a write to that generation, which should be accepted according to the changes in `cdc::metadata`. To fix this, we should store a list of committed generations in `system.topology`, not only the current one.

The second problem is that the topology coordinator (more specifically, the CDC generation publisher) may delete a generation from `cdc_generations_v3` too early. Currently, a generation can be removed from `cdc_generations_v3` if it's not a current generation and its timestamp is lower than `now - 24h`. However, after #17134, a write to such a generation can be accepted if  its the write's timestamp is greater than `now - generation_leeway`. In such a case, removing this generation is incorrect. We need to change the logic responsible for clearing `cdc_generations_v3`. For example, we can remove generations that **finished** operating earlier than `now - 24h`. This logic can also be applied to clear the list of committed generations mentioned above.
